devito/passes/clusters/aliases.py

```diff
@@ -490,7 +490,7 @@ def process(cluster, chosen, aliases, sregistry, platform):
 
         # Finally, build a new Cluster for `alias`
         built = cluster.rebuild(exprs=expression, ispace=ispace, dspace=dspace)
-        clusters.append(built)
+        clusters.insert(0, built)
```

Comment:
this is the actual fix -- I was breaking `Aliases.iter`'s interface (see docstring) 🤦